### PR TITLE
Add More Tests to the Test Suite

### LINKED ISSUE #203: Add tests to the game for quality assurance
**User Story**
As a developer, I'd like to add tests to our project, so that we're always sure it works

**Acceptance Criteria**

- More tests are added to the project

**Ideas**
- Dice rolls calculated
- Potion bonus
- Modifier calculated correctly
- Attack values (fail/succeed when they should)

### PR DESCRIPTION
**Pull request by Pyxxil**
GitHub Issue (If applicable): #203 

## PR Type

What kind of change does this PR introduce?
 - Tests

## What is the current behavior?
There are some tests in the test suite

## What is the new behavior?
More tests have been added, especially some which test parts of code that were introduced in this project.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [ ] Checked licensing of Frameworks/Libraries (if applicable) 

Internal Issue (If applicable): closes #203

### CODE REVIEW

File: src/__tests__/utils/calculate-dice.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        const notation = '1d4';
+        const range = calculateDamageRange(notation, true);
+        // The above should give the same result as this
+        const criticalHitNotation = '2d4';
+        const critRange = calculateDamageRange(criticalHitNotation, false);
+
+        expect(range).toEqual(critRange);
+    });
+
+    test('1d4 + 3 should be have the same range as 2d4 + 3', () => {
```
Comment (Matteas-Eden) [inline src/__tests__/utils/calculate-dice.js]: Typo here. Also, I'm not entirely sure this statement is true, do you mind explaining?

**Review by Matteas-Eden — CHANGES_REQUESTED** [review]
Looks mostly good, I just have a small thing I want clarified.

File: src/__tests__/utils/calculate-dice.js (hunk 1 above)
Comment (Pyxxil) [reply src/__tests__/utils/calculate-dice.js]: Your Issue for Crit's said that if you have the dice '1d4 + 2', then it should actually roll '2d4 + 2'. Hence, if we normally have the dice '1d4 + 3' then we should, on a critical hit, have the same range as '2d4 + 3'.

**Review by Matteas-Eden — APPROVED** [review]
Looks good!

**Pull request merged**